tools/diff-processor/breaking_changes/breaking_changes.go

```diff
@@ -43,9 +43,10 @@ func ComputeBreakingChanges(schemaDiff diff.SchemaDiff) []BreakingChange {
 			}
 		}
 
+		isPreExistingResource := !schemaDiff.IsNewResource(resource)
 		for field, fieldDiff := range resourceDiff.Fields {
 			for _, rule := range FieldDiffRules {
-				for _, message := range rule.Messages(resource, field, fieldDiff) {
+				for _, message := range rule.Messages(resource, field, fieldDiff, isPreExistingResource) {
```

Comment:
Rather than passing a bool isPreExistingResource, what would you think of passing the entire resourceDiff object? That would be a simple change that would give us more flexibility for potentially introducing more complex logic in the future.

Reply:
changed to pass interface, would rather not pass entire object, difficult to unit test